students/eowyn/session07/html-rendering/html_render.py

```
@@ -59,12 +84,73 @@ class OneLineTag(Element):
     def render(self, out_file, current_ind=""):
         # Pretty-print content and tags to file on one line.
         # Print the string representation of all objects within content.
-
-        out_file.write('\n' + current_ind + '<' + self.tag + '>')
+        self.render_open_tag(out_file, current_ind)
         for each in self.content:
             out_file.write(' ' + str(each) + ' ')
-        out_file.write('</' + self.tag + '>')
+        close_tag = '</{}>'.format(self.tag)
+        out_file.write(close_tag)
 
 
 class Title(OneLineTag):
     tag = 'title'
+
+
+class SelfClosingTag(Element):
+    # render tags like <hr /> and <br />
+    # render just one tag and any attributes
+
+    def render(self, out_file, current_ind=""):
+        # Render just a tag and any attributes, ignore contents
```

nice! but maybe you should raise an error if there are any contents -- probably in the __init__ -- that would be better than simply ignoring it.